_posts/2018-03-29-timmie.md

```diff
@@ -0,0 +1,38 @@
+author-email: joy.sangsrichan@gmail.com
+author-social:
+  github: https://github.com/jxs2143
+  linkedin: https://www.linkedin.com/in/joy-sangsrichan-390518150/
+---
+Meet Timmie, a discord bot that manages video game events and sends random dog pictures when requested! There is actually a story that Timmie is a robot who believes that he is a knight, hence why he is seen wearing a full Medieval-style armor. 
+
+#### What is Discord?
+Discord is an application that supports voice call, text chat, and screen-sharing designed for gaming communities. One of its features is the ability to add custom bots to the server such as welcome bots, music bots, and administrator bots. With the API wrapper, Discord.py, created by a user on Github named Rapptz, anyone can design and create their own bots. 
+
+#### What can Timmie do?
+Timmie's functions can be separated into four different categories: basic interactions, calculation, event management, and miscellaneous. You can view the full list of functions here on my Github page. 
+
+##### Basic Interactions
+The way that Timmie gets user input is through the function on_message(message) that reads directly from the server. Then, the bot checks if the message object begins with specific words such as "!hello" before returning a response which is an instance of the Discord message class. 
+
+##### Calculation
+All math commands follow the same input format { !command number operator number } ex. {!add 5 + 4}. The message is then parsed and split into a list of Strings before doing the calculation. If the user enters an invalid command, Timmie will return an error message. 
+
+##### Event Management
+Here, I used the Python class to create events, which are stored in a list in the bot file. In order to create a new event, the user must specify an event name, a description, the name of the game, and the maximum number of players. Once an event is created, any users can add/remove players, view all events, and remove an event from the list.
+
+##### Miscellaneous
+All miscellaneous commands follow the same input format { !command } ex. !dog. My favorite command in this category would be !dog, which returns a random photo of a dog from https://random.dog/. For that specific command, I used BeautifulSoup4 to look through the website html5 file and return the first image tag. 
+
+The most challenging part of this project is to experiment with the Discord API wrapper. There are not many online tutorials, which require you to carefully read through the documentation and be creative with how you approach the problem. Along the way, I have picked up some new Python skills; for instance, using pip and BeautifulSoup4. Our Opcomm director, Steven Mirabito, showed me how to host the bot on OpenShift, so it is now running on my high school eSports server. In the future, I plan to add more administrator functions and a scoreboard to keep track of all the matches.
```

Comment:
`s/is/was/` or `s/is/has been/` in your first sentence.
"...which means you are required to carefully read..."